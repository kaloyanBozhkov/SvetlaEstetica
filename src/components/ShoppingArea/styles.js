import { pagePaddings } from "~/theme/theme"

const link = {
    fontSize: '4.8rem',
    lineHeight: '5.6rem',
    color: 'white',
    fontWeight: 'bold',
    opacity: '0.2',
    bg: 'none',
    h: 'auto',
    p: '0',
    _hover: {
        color: 'text.300'
    },
    _active: {
        bg: 'none'
    },
}

const styles = {
    common: {
        height: '100%',
        transition: 'background .4s',
        ...pagePaddings
    },
    '/offers': {
        background: 'radial-gradient(50% 50% at 50% 50%, rgba(218, 171, 179, 0.5) 0%, rgba(255, 255, 255, 0) 100%), #D07081',
    },
    '/products': {
        background: 'radial-gradient(50% 50% at 50% 50%, rgba(250, 221, 225, 0.5) 0%, rgba(255, 255, 255, 0) 100%), #F486A8'
    },
    '/treatments': {
        background: 'radial-gradient(50% 50% at 50% 50%, rgba(219, 206, 232, 0.5) 0%, rgba(255, 255, 255, 0) 100%), #8675A9'
    },
    linkArea: {
        display: {
            xs: 'none',
            base: 'flex'
        },
        w: '100%',
        justifyContent: 'space-between',
        paddingTop: '3rem',
        paddingBottom: '3rem',
        overflow: 'hidden'
    },
    activeLink: {
        ...link,
        order: '-1',
        justifyContent: 'flex-start',
        flex: '1',
        opacity: 1
    },
    link: {
        ...link,
        marginLeft: '5rem',
        _hover: {
            opacity: .6,
        }
    }
}

export default styles