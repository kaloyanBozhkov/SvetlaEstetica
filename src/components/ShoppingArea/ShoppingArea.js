import { Box, Flex, Button, Text } from '@chakra-ui/core'
import React from 'react'
import Link from 'next/link'

import styles from './styles'


const paths = [
    {
        href: '/offers',
        as: '/offerte',
        text: 'Offerte'
    },
    {
        href: '/treatments',
        as: '/trattamenti',
        text: 'Trattamenti'
    },
    {
        href: '/products',
        as: '/prodotti',
        text: 'Prodotti'
    },
]

const ShoppingArea = ({ activePath, children }) => {
    return (
        <Box {...styles[activePath]} {...styles.common}>
            <Flex {...styles.linkArea}>
                {paths.map(({ href, as, text }, index) => (
                    <Link
                        key={index}
                        href={href}
                        as={as}
                        {...(activePath === href ? { 'data-test-id': 'activeLinkBtn'} : {})}
                    >
                        <Button {...(activePath === href ? styles.activeLink : styles.link)}>{text}</Button>
                    </Link>
                ))}
            </Flex>
            <Flex>
                
            </Flex>
            {children}
        </Box>
    )
}

export default ShoppingArea