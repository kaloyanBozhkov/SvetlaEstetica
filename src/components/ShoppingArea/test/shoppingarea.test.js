import { shallow } from 'enzyme'
import React from 'react'
import ShoppingArea from '../ShoppingArea'

describe('testing shopping area wrapper', () => {
    it('should render component with default props', () => {
        const wrapper = shallow(<ShoppingArea />)
        expect(wrapper).toMatchSnapshot()
    })

    it('should render children underneath the links area', () => {
        const wrapper = shallow(<ShoppingArea activePath="/products"><p>hello world</p></ShoppingArea>)
        expect(wrapper.find('p').text()).toEqual('hello world')
    })

    it('should render with active link for provided activePath', () => {
        const wrapper = shallow(<ShoppingArea activePath="/offers" />)
        expect(wrapper.find('[data-test-id="activeLinkBtn"]').prop('href')).toEqual('/offers')
    })
})