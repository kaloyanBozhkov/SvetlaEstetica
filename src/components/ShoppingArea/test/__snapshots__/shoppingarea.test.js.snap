// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing shopping area wrapper should render component with default props 1`] = `
<Box
  height="100%"
  paddingLeft={
    Object {
      "base": "50",
      "md": "8",
      "sm": "4",
      "xs": "2",
    }
  }
  paddingRight={
    Object {
      "base": "50",
      "md": "8",
      "sm": "4",
      "xs": "2",
    }
  }
>
  <Flex
    display={
      Object {
        "base": "flex",
        "xs": "none",
      }
    }
    justifyContent="space-between"
    paddingBottom="3rem"
    paddingTop="3rem"
    w="100%"
  >
    <Link
      as="/offerte"
      href="/offers"
      key="0"
    >
      <Button
        _hover={
          Object {
            "opacity": 0.6,
          }
        }
        bg="none"
        color="white"
        fontSize="4.8rem"
        fontWeight="bold"
        h="auto"
        lineHeight="5.6rem"
        marginLeft="5rem"
        opacity="0.2"
        p="0"
      >
        Offerte
      </Button>
    </Link>
    <Link
      as="/trattamenti"
      href="/treatments"
      key="1"
    >
      <Button
        _hover={
          Object {
            "opacity": 0.6,
          }
        }
        bg="none"
        color="white"
        fontSize="4.8rem"
        fontWeight="bold"
        h="auto"
        lineHeight="5.6rem"
        marginLeft="5rem"
        opacity="0.2"
        p="0"
      >
        Trattamenti
      </Button>
    </Link>
    <Link
      as="/prodotti"
      href="/products"
      key="2"
    >
      <Button
        _hover={
          Object {
            "opacity": 0.6,
          }
        }
        bg="none"
        color="white"
        fontSize="4.8rem"
        fontWeight="bold"
        h="auto"
        lineHeight="5.6rem"
        marginLeft="5rem"
        opacity="0.2"
        p="0"
      >
        Prodotti
      </Button>
    </Link>
  </Flex>
</Box>
`;
